librosa/core/spectrum.py

```diff
@@ -33,59 +33,89 @@
 @cache(level=20)
 def stft(y, n_fft=2048, hop_length=None, win_length=None, window='hann',
          center=True, dtype=np.complex64, pad_mode='reflect'):
-    """Short-time Fourier transform (STFT)
+    """Short-time Fourier transform (STFT). [1, chapter 2]
+    
+    The STFT represents a signal in the time-frequency domain by
+    computing discrete Fourier transforms (DFT) over short overlapping
+    windows.
 
-    Returns a complex-valued matrix D such that
+    This function returns a complex-valued matrix D such that
         `np.abs(D[f, t])` is the magnitude of frequency bin `f`
         at frame `t`
-
+    and
         `np.angle(D[f, t])` is the phase of frequency bin `f`
-        at frame `t`
+        at frame `t`.
+        
+    The integers `t` and `f` can be converted to physical units by means
+    of the utility functions `frames_to_sample` and `fft_frequencies`.
+        
+    .. [1] M. Müller. "Fundamentals of Music Processing." Springer, 2015
+         
 
     Parameters
     ----------
     y : np.ndarray [shape=(n,)], real-valued
-        the input signal (audio time series)
+        input signal
 
     n_fft : int > 0 [scalar]
-        FFT window size
+        length of the windowed signal after padding with zeros.
+        The number of rows in the STFT matrix `D` is (1 + n_fft/2).
+        We recommend setting `n_fft` to a power of two for optimizing
+        the speed of the fast Fourier transform algorithm.
 
     hop_length : int > 0 [scalar]
-        number audio of frames between STFT columns.
-        If unspecified, defaults `win_length / 4`.
+        number of audio samples between adjacent STFT columns.
+        Smaller values increase the number of columns in `D` without
+        affecting its temporal resolution.
```

Comment:
This doesn't seem right. I think you mean that it doesn't change the frequency resolution?

Reply:
I persist. It changes neither the temporal resolution nor the frequency resolution (Heisenbergly speaking).
This comment is meant to distinguish the effects of hop length and window length

Reply:
Yes, but the distinction here is between the input (sample) rate (unaffected) and the output (frame) rate (definitely affected).  In the current phrasing, it's ambiguous which one you're referring to.